src/scf/self_consistent_field.jl

```diff
@@ -181,42 +170,49 @@ Overview of parameters:
         Δρ = ρout - ρin
 
         # Update info with results gathered so far
-        info = (; ham, basis, converged, stage=:iterate, algorithm="SCF",
-                ρin, ρout, α=damping, n_iter, nbandsalg.occupation_threshold,
-                runtime_ns=time_ns() - start_ns, nextstate...,
-                diagonalization=[nextstate.diagonalization])
+        info_next = (; ham, basis, converged, stage=:iterate, algorithm="SCF",
+                       ρin, α=damping, n_iter, nbandsalg.occupation_threshold,
+                       runtime_ns=time_ns() - start_ns, nextstate...,
+                       diagonalization=[nextstate.diagonalization])
 
         # Compute the energy of the new state
         if compute_consistent_energies
             (; energies) = energy(basis, ψ, occupation; ρ=ρout, eigenvalues, εF)
         end
+        history_Etot = vcat(info.history_Etot, energies.total)
+        history_Δρ = vcat(info.history_Δρ, norm(Δρ) * sqrt(basis.dvol))
```

Comment:
I think it's ok to just push and risk that the `info.history_*` things are globally updated.

Reply:
Agree mostly, but I think avoiding this risk makes the state more transparent in distinguishing `info` from `info_next`.